lib/mail/version_specific/ruby_1_9.rb

```diff
@@ -125,7 +125,11 @@ def Ruby19.q_value_decode(str)
     def Ruby19.param_decode(str, encoding)
       str = uri_parser.unescape(str)
       str = charset_encoder.encode(str, encoding) if encoding
-      str
+      decoded = str.encode(Encoding::UTF_8, :invalid => :replace, :replace => "")
```

Comment:
I agree, I remember that we talked about this in https://github.com/mikel/mail/pull/978

My end goal is that `b_value_decode` `q_value_decode` and `param_decode` all use the same method to encode to UTF-8 and that they all use the unicode replacement car. We can get there using whatever path you'd like, but my idea was to:

1. add this to param_decode first
2. refactor the three implementations down to 1 method
3. change that one method to use the unicode replacement char

How do you feel about that approach? Do you have something different in mind?